### PR TITLE
Add vector mask testAny, testAll and conversion opcodes

### PR DESCRIPTION
**Pull request by gita-omr**
- the opcodes return true iff at least one element(testAny) or all
  elements(testALL) of the vecror mask are true
- both testAny and testAll opcodes have masked and non-masked
  versions, so add mTestAny, mTestAll, mmTestAny, mmTestAll

### CODE REVIEW

**Comment by gita-omr** [comment]
Jenkins build all

File: compiler/il/VectorOperations.enum [hunk 1]
```diff
@@ -18,6 +18,70 @@
...
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    test if all mask elements are true */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mmAnyTrue, \
+   /* .name                    = */ "mmAnyTrue", \
+   /* .properties1             = */ 0, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
```
Comment (BradleyWood) [inline compiler/il/VectorOperations.enum:58]: I think we need a flag to indicate that this operation is on a mask, under a mask

File: compiler/il/VectorOperations.enum [hunk 2]
```diff
@@ -18,6 +18,70 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ mAnyTrue, \
+   /* .name                    = */ "mAnyTrue", \
+   /* .properties1             = */ 0, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
```
Comment (BradleyWood) [inline compiler/il/VectorOperations.enum:26]: Do we need a flag to indicate these operations are on a mask (not under a mask)?

File: compiler/il/VectorOperations.enum (hunk 1 above)
Comment (gita-omr) [reply compiler/il/VectorOperations.enum:58]: Yes, I agree, will add.

File: compiler/il/VectorOperations.enum (hunk 2 above)
Comment (gita-omr) [reply compiler/il/VectorOperations.enum:26]: Agree

**Comment by gita-omr** [comment]
Rebased.

File: compiler/il/VectorOperations.enum (hunk 1 above)
Comment (gita-omr) [reply compiler/il/VectorOperations.enum:58]: Added `ILProp1::VectorMasked` property. But after some thought, not sure about "on mask" property. Usually we check the return type of the opcode vs the type of any of its children. Do you think we need that flag? @BradleyWood

**Comment by gita-omr** [comment]
Added VectorMasked property.

File: compiler/il/VectorOperations.enum (hunk 1 above)
Comment (BradleyWood) [reply compiler/il/VectorOperations.enum:58]: You're right, its not necessary. I leave it up to you

**Comment by knn-k** [comment]
The aarch64/arm part looks good to me.

**Comment by gita-omr** [comment]
Rebased

**Comment by gita-omr** [comment]
Added commit for mask conversion opcodes.

**Comment by gita-omr** [comment]
Fixed line endings

**Comment by gita-omr** [comment]
Jenkins build all

**Comment by gita-omr** [comment]
WIP since I would like to add some fixes to this PR.

**Comment by gita-omr** [comment]
Rebased.

**Comment by gita-omr** [comment]
Fixed a bug in getVectorOperation().

**Comment by knn-k** [comment]
Jenkins build all

**Comment by knn-k** [comment]
The following failure on linux ppc needs to be investigated in Issue #6571.

```
[2022-10-14T04:21:57.059Z] 30: [0;32m[----------] [m14 tests from Special/PPCDirectEncodingTest
[2022-10-14T04:21:57.059Z] 30: free(): invalid next size (normal)
[2022-10-14T04:21:57.059Z] 30/30 Test #30: compunittest ......................Child aborted***Exception:   0.75 sec
```

**Review by knn-k — APPROVED** [review]

**Pull request merged**